pkg/apis/integreatly/openshift/osclient.go

```diff
@@ -39,6 +40,25 @@ func (osClient *OSClient) Bootstrap(namespace string, kubeconfig *rest.Config) e
 	return nil
 }
 
+func (osClient *OSClient) GetDC(ns string, dcName string) (osappsv1.DeploymentConfig, error) {
+	dcs, err := osClient.ocDCClient.DeploymentConfigs(ns).List(meta_v1.ListOptions{})
```

Comment:
I think you can add a metadata filter in the Listoptions struct so you don't need to iterate over the list

Reply:
is there more than one dc? What is wrong with using ```osClient.ocDCClienty.DeploymentConfigs(ns).Get(dcName)```